cmd/frontend/graphqlbackend/search_results_test.go

```diff
@@ -39,33 +41,25 @@ import (
 
 var mockCount = func(_ context.Context, options database.ReposListOptions) (int, error) { return 0, nil }
 
-func assertEqual(t *testing.T, got, want interface{}) {
-	t.Helper()
-
-	if diff := cmp.Diff(got, want); diff != "" {
-		t.Fatalf("(-want +got):\n%s", diff)
-	}
-}
-
 func TestSearchResults(t *testing.T) {
 	if os.Getenv("CI") != "" {
 		// #25936: Some unit tests rely on external services that break
 		// in CI but not locally. They should be removed or improved.
-		t.Skip("TestSeachResults only works in local dev and is not reliable in CI")
+		t.Skip("TestSearchResults only works in local dev and is not reliable in CI")
 	}
-	db := new(dbtesting.MockDB)
 
-	limitOffset := &database.LimitOffset{Limit: search.SearchLimits(conf.Get()).MaxRepos + 1}
+	ctx := context.Background()
+	db := dbmock.NewMockDB()
+
+	wantLimitOffset := &database.LimitOffset{Limit: search.SearchLimits(conf.Get()).MaxRepos + 1}
 
 	getResults := func(t *testing.T, query, version string) []string {
-		r, err := (&schemaResolver{db: database.NewDB(db)}).Search(context.Background(), &SearchArgs{Query: query, Version: version})
-		if err != nil {
-			t.Fatal("Search:", err)
-		}
-		results, err := r.Results(context.Background())
-		if err != nil {
-			t.Fatal("Results:", err)
-		}
+		r, err := newSchemaResolver(db).Search(ctx, &SearchArgs{Query: query, Version: version})
+		require.Nil(t, err)
+
+		results, err := r.Results(ctx)
+		require.Nil(t, err)
```

Comment:
nit: `require.NoError(t, err)` gives a nicer error message on failure

```suggestion
		require.NoError(t, err)
```